src/ng-select/lib/ng-select.component.ts

```diff
@@ -515,7 +522,17 @@ export class NgSelectComponent implements OnDestroy, OnChanges, AfterViewInit, C
         return empty && this._isTypeahead && !this.searchTerm && !this.loading;
     }
 
+    handleSearchComposition(isCompositionStart: boolean, term: string) {
+        this._IMEComposition = isCompositionStart;
+        if (this._IMEComposition === false) {
+            this.filter(term);
+        }
+    }
+
     filter(term: string) {
+        if (this._IMEComposition) {
+            return;
```

Comment:
this condition probably needs to be added to other cases also - when the user presses `enter` or `tab` it should skip until the composition is ended.

Reply:
I did some tests, browser will not receive `tab` or `enter` during compotion.